### PR TITLE
Add functional test covering an `IList<T>` property

### PR DESCRIPTION
**Pull request by dougbu**
- #5656

### CODE REVIEW

**Comment by dougbu** [comment]
FYI Travis failure is same `IOException` we see in `dev` branch far too often (#6033).

File: test/WebSites/HtmlGenerationWebSite/Views/HtmlGeneration_Home/Order.cshtml [hunk 1]
```diff
@@ -34,6 +34,34 @@
...
+                        </th>
+                        <th>
+                            <label asp-for="ProductDetails.FirstOrDefault().ProductName"></label>
+                        </th>
+                    </tr>
+                    @for (var i = 0; i < Model.ProductDetails.Count; i++)
+                    {
+                    <tr>
+                        <td>
+                            <input asp-for="ProductDetails[i].Number" />
```
Comment (NTaylorMullen) [inline test/WebSites/HtmlGenerationWebSite/Views/HtmlGeneration_Home/Order.cshtml]: Does this sufficiently fix the issue? Given that you're iterating over the list our `TagHelper`s only ever see the `int` property Number, not the actual List.

File: test/WebSites/HtmlGenerationWebSite/Views/HtmlGeneration_Home/OrderUsingHtmlHelpers.cshtml [hunk 2]
```diff
@@ -28,6 +28,34 @@
...
+                        </th>
+                        <th>
+                            @Html.LabelFor(m => m.ProductDetails.FirstOrDefault().ProductName)
+                        </th>
+                    </tr>
+                    @for (var i = 0; i < Model.ProductDetails.Count; i++)
+                    {
+                        <tr>
+                            <td>
+                                @Html.TextBoxFor(m => m.ProductDetails[i].Number, htmlAttributes: new { type = "number" })
```
Comment (NTaylorMullen) [inline test/WebSites/HtmlGenerationWebSite/Views/HtmlGeneration_Home/OrderUsingHtmlHelpers.cshtml]: Same Q

File: test/WebSites/HtmlGenerationWebSite/Views/HtmlGeneration_Home/Order.cshtml (hunk 1 above)
Comment (NTaylorMullen) [inline test/WebSites/HtmlGenerationWebSite/Views/HtmlGeneration_Home/Order.cshtml]: Talked offline. This is good 👍

**Comment by dougbu** [comment]
1672820e4

**Pull request closed without merging**